Sources/StreamChat/Controllers/DatabaseObserver/BackgroundDatabaseObserver.swift

```diff
@@ -34,22 +34,33 @@ class BackgroundDatabaseObserver<Item, DTO: NSManagedObject> {
     private var _items: [Item] = []
 
     /// The items that have been fetched and mapped
+    ///
+    /// -Note: Fetches items synchronously if the initial fetch has not finished.
     var rawItems: [Item] {
-        queue.sync { _items }
+        // If we have already loaded initial items, return them immediately
+        if let items = queue.sync(execute: { _hasFetchedInitialItems ? _items : nil }) {
+            return items
+        }
+        // Otherwise load items synchronously if the initial fetch has not finished
+        return queue.sync(flags: .barrier) {
+            var items = [Item]()
+            frc.managedObjectContext.performAndWait {
+                items = mapItems(changes: nil, reusableItems: [])
+            }
+            _items = items
+            _hasFetchedInitialItems = true
+            return _items
+        }
```

Comment:
We should be able to remove this now `BackgroundListDatabaseObserver.refreshItems()`, I think it makes sense to be part of this PR, since the `refreshItems()` was introduced because of this limitation. After that, we should check that the Channel List and Thread List states work correctly. (Loading View, Empty View and Error View)

Reply:
I'll review it

Reply:
Done, removed